homeassistant/components/overkiz/cover_entities/generic_cover.py

```diff
@@ -0,0 +1,216 @@
+    CoverEntity,
+)
+from homeassistant.components.overkiz.entity import OverkizEntity
+
+ATTR_OBSTRUCTION_DETECTED = "obstruction-detected"
+
+COMMANDS_STOP: list[OverkizCommand] = [
+    OverkizCommand.STOP,
+    OverkizCommand.STOP_IDENTIFY,
+    OverkizCommand.MY,
+]
+COMMANDS_STOP_TILT: list[OverkizCommand] = [
+    OverkizCommand.STOP,
+    OverkizCommand.STOP_IDENTIFY,
+    OverkizCommand.MY,
+]
+COMMANDS_OPEN: list[OverkizCommand] = [
+    OverkizCommand.OPEN,
+    OverkizCommand.UP,
+    OverkizCommand.CYCLE,
+]
+COMMANDS_OPEN_TILT: list[OverkizCommand] = [OverkizCommand.OPEN_SLATS]
+COMMANDS_CLOSE: list[OverkizCommand] = [
+    OverkizCommand.CLOSE,
+    OverkizCommand.DOWN,
+    OverkizCommand.CYCLE,
+]
+COMMANDS_CLOSE_TILT: list[OverkizCommand] = [OverkizCommand.CLOSE_SLATS]
+
+COMMANDS_SET_TILT_POSITION: list[OverkizCommand] = [OverkizCommand.SET_ORIENTATION]
+
+
+class OverkizGenericCover(OverkizEntity, CoverEntity):
+    """Representation of a TaHoma Cover."""
+
+    @property
+    def current_cover_tilt_position(self) -> int | None:
+        """Return current position of cover tilt.
+
+        None is unknown, 0 is closed, 100 is fully open.
+        """
+        if position := self.executor.select_state(
+            OverkizState.CORE_SLATS_ORIENTATION, OverkizState.CORE_SLATE_ORIENTATION
+        ):
+            return 100 - cast(int, position)
+
+        return None
+
+    async def async_set_cover_tilt_position(self, **kwargs: Any) -> None:
+        """Move the cover tilt to a specific position."""
+        if command := self.executor.select_command(*COMMANDS_SET_TILT_POSITION):
+            await self.executor.async_execute_command(
+                command,
+                100 - kwargs.get(ATTR_TILT_POSITION, 0),
+            )
+
+    @property
+    def is_closed(self) -> bool | None:
+        """Return if the cover is closed."""
+
+        state = self.executor.select_state(
+            OverkizState.CORE_OPEN_CLOSED,
+            OverkizState.CORE_SLATS_OPEN_CLOSED,
+            OverkizState.CORE_OPEN_CLOSED_PARTIAL,
+            OverkizState.CORE_OPEN_CLOSED_PEDESTRIAN,
+            OverkizState.CORE_OPEN_CLOSED_UNKNOWN,
+            OverkizState.MYFOX_SHUTTER_STATUS,
+        )
+        if state is not None:
+            return state == OverkizCommandParam.CLOSED
+
+        # Keep this condition after the previous one. Some device like the pedestrian gate, always return 50 as position.
+        if self.current_cover_position is not None:
+            return self.current_cover_position == 0
+
+        if self.current_cover_tilt_position is not None:
+            return self.current_cover_tilt_position == 0
+
+        return None
+
+    async def async_open_cover_tilt(self, **kwargs: Any) -> None:
+        """Open the cover tilt."""
+        if command := self.executor.select_command(*COMMANDS_OPEN_TILT):
+            await self.executor.async_execute_command(
+                self.executor.select_command(command)
```

Comment:
@epenet would you have an idea why this mypy error is present? If I cast command to str on line 99, it complains that this is a redundant cast. 

`Argument 1 to "async_execute_command" of "OverkizExecutor" has incompatible type "Optional[str]"; expected "str"  [arg-type]`

Reply:
Maybe you need to cast the select_command, not the command:
`cast(str, self.executor.select_command(command))`

Reply:
That select_command is redundant as well... Thanks!